x-pack/plugins/log_explorer/public/components/common/popover_chip.tsx

```diff
@@ -0,0 +1,145 @@
+  /**
+   * ECS mapping for the key
+   */
+  property: string;
+  /**
+   * Value for the mapping, which will be displayed
+   */
+  text: string;
+  dataTestSubj?: string;
+  leftSideIcon?: EuiBadgeProps['iconType'];
+  rightSideIcon?: EuiBadgeProps['iconType'];
+  borderColor?: string | null;
+  style?: React.CSSProperties;
+  shouldRenderPopover?: boolean;
+}
+
+const convertStyleToString = (style: React.CSSProperties) => {
+  return Object.entries(style)
+    .map(([key, value]) => `${key}: ${value};`)
+    .join(' ');
+};
+
+export function ChipWithPopover({
+  property,
+  text,
+  dataTestSubj = `${property}_${text.replace(/ +/g, '')}`,
+  leftSideIcon,
+  rightSideIcon,
+  borderColor,
+  style,
+  shouldRenderPopover = true,
+}: ChipWithPopoverProps) {
+  const xsFontSize = useEuiFontSize('xs').fontSize;
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const handleChipClick = useCallback(() => {
+    if (!shouldRenderPopover) return;
+    setIsPopoverOpen(true);
+  }, [shouldRenderPopover]);
+
+  const closePopover = () => setIsPopoverOpen(false);
+
+  const chipContent = (
+    <EuiBadge
+      color="hollow"
+      iconType={rightSideIcon}
+      iconSide="right"
+      data-test-subj={`dataTablePopoverChip_${dataTestSubj}`}
+      onClick={handleChipClick}
+      onClickAriaLabel={openCellActionPopoverAriaText}
+      css={css`
+        ${borderColor ? `border: 2px solid ${borderColor};` : ''}
+        font-size: ${xsFontSize};
+        display: flex;
+        justify-content: center;
+        ${style ? css(convertStyleToString(style)) : ''}
+      `}
+    >
+      <EuiFlexGroup gutterSize="xs">
+        {leftSideIcon && (
+          <EuiFlexItem>
+            <EuiIcon type={leftSideIcon} />
+          </EuiFlexItem>
+        )}
+        <EuiFlexItem>{text}</EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiBadge>
+  );
+
+  return (
+    <EuiPopover
+      button={chipContent}
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+      anchorPosition="downCenter"
+      panelPaddingSize="s"
+      panelStyle={{ minWidth: '24px' }}
+    >
+      <EuiFlexGroup
+        gutterSize="none"
+        direction="row"
+        responsive={false}
+        data-test-subj="dataTableCellActionPopoverTitle"
+      >
+        <EuiFlexItem>
+          <div style={{ maxWidth: '200px' }}>
+            <EuiText size="s">
+              <span className="unifiedDataTable__cellPopoverValue eui-textBreakWord">
+                {property}:{text}
+              </span>
+            </EuiText>
+          </div>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButtonIcon
+            aria-label={closeCellActionPopoverText}
+            data-test-subj="dataTableExpandCellActionPopoverClose"
+            iconSize="s"
+            iconType="cross"
+            size="xs"
+            onClick={closePopover}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiPopoverFooter>
```

Comment:
Just a thought: At some point it might be worth re-implementing this based on the UiActions mechanism that allows for registering additional triggers. But this is probably fine for the initial version.